pkg/api/api.go

```diff
@@ -180,6 +183,18 @@ func requestEncrypt(r *http.Request) bool {
 	return strings.ToLower(r.Header.Get(SwarmEncryptHeader)) == "true"
 }
 
+func requestPostageBatchId(r *http.Request) ([]byte, error) {
+	if h := strings.ToLower(r.Header.Get(SwarmPostageBatchIdHeader)); h != "" {
+		if len(h) != 64 {
+			return nil, errInvalidPostageBatch
+		}
+		return hex.DecodeString(h)
```

Comment:
If value in the header is not hexadecimal it will return some internal error. This might have been handled, and returned `errInvalidPostageBatch` again.

Reply:
makes sense :+1:  thanks